controllers/buildpipeline/buildpipeline_adapter_test.go

```diff
@@ -290,7 +293,21 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 	})
 
 	AfterEach(func() {
-		err := k8sClient.Delete(ctx, buildPipelineRun)
+		patch := client.MergeFrom(buildPipelineRun.DeepCopy())
+		_ = controllerutil.RemoveFinalizer(buildPipelineRun, "build.appstudio.openshift.io/pipelinerun")
+		err := k8sClient.Patch(ctx, buildPipelineRun, patch)
```

Comment:
I'm running into an issue here.  It looks like the finalizer isn't being removed, which is preventing this deletion from being completed.  As far as I can tell, lines 296-298 are the same (minus error handling) as the ones I've added in `removeFinalizerAndContinueProcessing()`.